packages/core/src/interfaces/axis-scales.ts

```diff
@@ -22,6 +23,7 @@ export interface AxisOptions {
 	 * Range usually follows a linear scale
 	 */
 	useAsRange?: boolean;
+	domain?: AxisDomain[];
```

Comment:
what's the definition of the `AxisDomain` type? Domains in d3 are usually an array of 2 values, is that reflected within that type?

Reply:
This is the definition:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/84a9f0a15d5e765e20fef3a04b43824c9d3370e1/types/d3-axis/index.d.ts#L21

I think you are right, we could make it a tuple of 2 elements, it would be more specific.

This makes me think of a problem: there is a naming conflict, `domain` is something else in your code, here we mean a "Scale Domain". How would you call it?

Reply:
hey @theiliad,  here's the definition: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/84a9f0a15d5e765e20fef3a04b43824c9d3370e1/types/d3-axis/index.d.ts#L21

Actually d3 domains are arrays of 2 or more values, as you can see here with the piecewise scale for example: https://github.com/d3/d3-scale#continuous_domain
`AxisDomain` type is just a helper type to define all the possibile values of the domain.

Open to discuss if you think it would be better to restrict the definition of the domain setting in this case though, as cesare was suggesting. But in that case I think it would be better to name it differently, to differentiate it from the "standard" d3 scale domain which may contain more than 2 values

Reply:
Right good catch 👍